runtime/vm/ContinuationHelpers.cpp

```diff
@@ -1207,8 +1207,8 @@ takeVirtualThreadListToUnblock(J9VMThread *currentThread)
 					continue;
 				case JAVA_LANG_VIRTUALTHREAD_WAIT:
 				case JAVA_LANG_VIRTUALTHREAD_TIMED_WAIT:
-					/* WAIT/TIMED_WAIT can only be added to blocked list if they have been notifed and threads updated. */
-					Assert_VM_unreachable();
+					/* WAIT/TIMED_WAIT can only be added to blocked list if they have been notified. */
+					Assert_VM_true(J9VMJAVALANGVIRTUALTHREAD_NOTIFIED(currentThread, current->vthread));
```

Comment:
if the assert is disabled, `J9VMJAVALANGVIRTUALTHREAD_NOTIFIED` won't be invoked. so, it should be invoked separately before `Assert_VM_true`.

Reply:
it is a getter used by the assert, we shouldn't need it if assert is disabled

Reply:
Confirmed locally and via grinder the assertion issue is fixed, now `java/lang/Thread/virtual/CancelTimerWithContention.java` fails due to hang/timeout.
I will try to apply the FLC bit patch and see if that fixes the new issue.